examples/simple_talker/simple_talker.c

```diff
@@ -292,19 +293,21 @@ int gptpdeinit(int *igb_shm_fd, char *igb_mmap)
 	return 0;
 }
 
-int gptpscaling(char *igb_mmap, gPtpTimeData *td)
+int gptpscaling(const char *igb_mmap, gPtpTimeData *td)
 {
 	if (NULL == td)
 		return -1;
 
-	pthread_mutex_lock((pthread_mutex_t *) igb_mmap);
-	memcpy(td, igb_mmap + sizeof(pthread_mutex_t), sizeof(*td));
-	pthread_mutex_unlock((pthread_mutex_t *) igb_mmap);
+	(void) pthread_mutex_lock((pthread_mutex_t *) igb_mmap);
+	{
+		memcpy(td, igb_mmap + sizeof(pthread_mutex_t), sizeof(*td));
```

Comment:
Hey,

as you noticed, this was simply done to highlight this critical section. It
is a common pattern to see its start and end.
Am 04.02.2015 18:18 schrieb "Levi Pearson" <notifications@github.com>:

> In examples/simple_talker/simple_talker.c
> <https://github.com/AVnu/Open-AVB/pull/185#discussion_r24100390>:
>
> >  {
> >  	if (NULL == td)
> >  		return -1;
> >
> > -	pthread_mutex_lock((pthread_mutex_t *) igb_mmap);
> > -	memcpy(td, igb_mmap + sizeof(pthread_mutex_t), sizeof(*td));
> > -	pthread_mutex_unlock((pthread_mutex_t *) igb_mmap);
> > +	(void) pthread_mutex_lock((pthread_mutex_t *) igb_mmap);
> > +	{
> > +		memcpy(td, igb_mmap + sizeof(pthread_mutex_t), sizeof(*td));
>
> I don't have a problem with this, but I'm curious whether it makes a
> functional difference in the code that's generated or if it was just done
> to highlight the mutex-protected function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AVnu/Open-AVB/pull/185/files#r24100390>.
>